rfcs/text/0428-extending-types.md

```diff
@@ -0,0 +1,677 @@
+
+**An example of _disjoint_ properties**:
+
+- Supertype `Person` contains required properties `Name` and `Age`
+- Supertype `Superhero` contains the property `Superpower`
+
+In this example, there is no overlap between properties, so an `Employee` type could have `Person` and `Superhero` as supertypes
+
+```txt
+              (supertypes)
+superpower◄────Superhero──┐
+                          │
+      name◄┐              │
+           ├───Person─────Employee (subtype)
+       age◄┘              │
+                          │
+occupation◄───────────────┘
+```
+
+**An example of _compatible_, overlapping properties**:
+
+- Supertype `Person` contains the required properties `Name` and `Age`
+- Supertype `Superhero` contains the required properties `Superpower` and `Name`
+
+In this example, `Name` overlaps as a required property in both supertypes.
+
+```txt
+              (supertypes)
+superpower◄────Superhero──┐
+               │          │
+      name◄────┤          │
+               │          │
+       age◄────Person─────Employee (subtype)
+                          │
+occupation◄───────────────┘
+
+
+```
+
+**An example of _incompatible_, overlapping properties**:
+
+- Supertype `Person` contains the required properties `Name` and `Age`
+- Supertype `Superhero` contains the required property `Superpower` and an array of `Name`s
+
+In this example, the array of `Name`s on the `Superhero` type would not be compatible with the required `Name` property of `Person`, which means that the two types cannot be supertypes together.
+
+## Additional properties on types
+
+In the proposed [Versioning RFC](./0408-versioning-types.md) for the type system, having `{ "additionalProperties": false }` for all schemas is an assumption made for determining type compatibility, which means that any supertype will not validate against a subtype that adds properties if it receives all properties of a subtype instance. For example, if we supply the `Employee` instance from above to a `Person`, it will receive properties that are considered `additionalProperties` (the `Occupation` property is not present on `Person`).
+
+The assumption that we can select/project parts of a subtype that make up a supertype is essential for keeping strictness in JSON Schemas.
+
+And unfortunately, specifying `{ "unevaluatedProperties": false }` does not behave as expected when composing types together in JSON Schema either, which means we will have to redefine how `{ "additionalProperties": true }` or `{ "unevaluatedProperties": false }` behaves within our type extension system to make supertypes keep strictness while allowing composition.
+
+Concrete examples of how JSON Schema breaks with these validation constraints are shown in the [Reference-level explanation](#additional-properties-on-types-1)
```

Comment:
This anchor does not exist

Reply:
Fixed in ffa59d98562a7a2129e16e1a0997a00bd8d2bdae, this was slightly related to the other comment about `unevaluatedProperties` explanation. Thanks for flagging :)